internal/toolexec/processors/aspect/onlink.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package aspect
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/datadog/orchestrion/internal/toolexec/processors"
+	"github.com/datadog/orchestrion/internal/toolexec/processors/aspect/linkdeps"
+	"github.com/datadog/orchestrion/internal/toolexec/proxy"
+)
+
+func (w Weaver) OnLink(cmd *proxy.LinkCommand) error {
+	reg, err := processors.ParseImportConfig(cmd.Flags.ImportCfg)
+	if err != nil {
+		return fmt.Errorf("parsing %q: %w", cmd.Flags.ImportCfg, err)
+	}
+
+	var changed bool
+	for _, archive := range reg.PackageFile {
+		data, err := archiveData(archive, linkdeps.LinkDepsFilename)
+		if err != nil {
+			return fmt.Errorf("reading %s from %q: %w", linkdeps.LinkDepsFilename, archive, err)
+		} else if data == nil {
+			// No link dependencies registered here...
+			continue
+		}
+
+		linkDeps, err := linkdeps.Read(data)
+		if err != nil {
+			return fmt.Errorf("reading %s from %q: %w", linkdeps.LinkDepsFilename, archive, err)
+		}
+
+		for _, depPath := range linkDeps.Dependencies() {
+			deps, err := resolvePackageFiles(depPath)
+			if err != nil {
+				return fmt.Errorf("resolving %q: %w", depPath, err)
+			}
+			for p, a := range deps {
+				if _, found := reg.PackageFile[p]; !found {
+					reg.PackageFile[p] = a
+					changed = true
+				}
+			}
+		}
+	}
+
+	if !changed {
+		return nil
+	}
+
+	if err := os.Rename(cmd.Flags.ImportCfg, cmd.Flags.ImportCfg+".original"); err != nil {
+		return fmt.Errorf("renaming %q: %w", cmd.Flags.ImportCfg, err)
+	}
+	file, err := os.Create(cmd.Flags.ImportCfg)
+	if err != nil {
+		return fmt.Errorf("creating %q: %w", cmd.Flags.ImportCfg, err)
+	}
+	defer file.Close()
+
+	if _, err := reg.WriteTo(file); err != nil {
+		return fmt.Errorf("writing updated %q: %w", cmd.Flags.ImportCfg, err)
+	}
+
+	return nil
+}
+
+func archiveData(archive, entry string) (io.Reader, error) {
```

Comment:
Nit: this func name kinda got me confused while reading the code since my first thought was that it actually would archive some data. `dataFromArchive()` or `readArchiveData()` may be a bit more explicit.
Or a short comment.

Reply:
Good point... I think `read` prefix would help indeed!